#!/usr/bin/env python
PACKAGE = "pyramid_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("winch_1_x", double_t, 0, "winch_1_x",  5.0, 0.0, 5.0)
gen.add("winch_1_y", double_t, 0, "winch_1_y",  5.0, 0.0, 5.0)
gen.add("winch_1_z", double_t, 0, "winch_1_z",  0.0, 0.0, 5.0)

gen.add("winch_2_x", double_t, 0, "winch_2_x", -5.0, -5.0, 0.0)
gen.add("winch_2_y", double_t, 0, "winch_2_y",  5.0,  0.0, 5.0)
gen.add("winch_2_z", double_t, 0, "winch_2_z",  0.0,  0.0, 5.0)

gen.add("winch_3_x", double_t, 0, "winch_3_x", -5.0, -5.0, 0.0)
gen.add("winch_3_y", double_t, 0, "winch_3_y", -5.0, -5.0, 0.0)
gen.add("winch_3_z", double_t, 0, "winch_3_z",  0.0,  0.0, 5.0)

gen.add("winch_4_x", double_t, 0, "winch_4_x",  5.0,  0.0, 5.0)
gen.add("winch_4_y", double_t, 0, "winch_4_y", -5.0, -5.0, 0.0)
gen.add("winch_4_z", double_t, 0, "winch_4_z",  0.0,  0.0, 5.0)

gen.add("lambda_1", double_t, 0, "Sliding surface parameter 1", 1.0, 0, 10.0)
gen.add("lambda_2", double_t, 0, "Sliding surface parameter 2", 1.0, 0, 10.0)
gen.add("lambda_3", double_t, 0, "Sliding surface parameter 3", 1.0, 0, 10.0)
gen.add("lambda_4", double_t, 0, "Sliding surface parameter 4", 1.0, 0, 10.0)
gen.add("lambda_5", double_t, 0, "Sliding surface parameter 5", 1.0, 0, 10.0)
gen.add("lambda_6", double_t, 0, "Sliding surface parameter 6", 1.0, 0, 10.0)

gen.add("K_1", double_t, 0, "Gain parameter 1", 2.0, 0, 10.0)
gen.add("K_2", double_t, 0, "Gain parameter 2", 2.0, 0, 10.0)
gen.add("K_3", double_t, 0, "Gain parameter 3", 2.0, 0, 10.0)
gen.add("K_4", double_t, 0, "Gain parameter 4", 2.0, 0, 10.0)
gen.add("K_5", double_t, 0, "Gain parameter 5", 2.0, 0, 10.0)
gen.add("K_6", double_t, 0, "Gain parameter 6", 2.0, 0, 10.0)

exit(gen.generate(PACKAGE, "pyramid_control", "SlidingModeController"))
